jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-client/src/main/java/org/jbpm/workbench/cm/client/roles/CaseRolesViewImpl.java

```diff
@@ -129,23 +157,21 @@ public void setupPagination() {
     public void setVisibleItems(List visibleItems) {
         removeAllChildren(roles);
         int visibleItemsSize = visibleItems.size();
-        if(visibleItemsSize>0){
-            visibleItems.stream().forEach(e -> ((CaseRoleItemView)e).setLastElementStyle(false));
-            ((CaseRoleItemView)visibleItems.get(visibleItemsSize-1)).setLastElementStyle(true);
+        if (visibleItemsSize > 0) {
+            visibleItems.forEach(e -> ((CaseRoleItemView) e).setLastElementStyle(false));
+            ((CaseRoleItemView) visibleItems.get(visibleItemsSize - 1)).setLastElementStyle(true);
         }
-        visibleItems.stream().forEach(e -> roles.appendChild(((CaseRoleItemView)e).getElement()) );
+        visibleItems.forEach(e -> roles.appendChild(((CaseRoleItemView) e).getElement()));
     }
 
     @Override
     public Div getScrollBox() {
         return scrollbox;
     }
 
-    @EventHandler("user-add")
-    public void onUserAddClick(@ForEvent("click") Event e) {
-        if (userAddCommand != null) {
-            userAddCommand.execute();
-        }
+    @EventHandler("filter-select")
+    public void onRolesAssignmentFilterChange(@ForEvent("change") Event e) {
+        filterCommand.execute();
```

Comment:
Does @ForEvent("change") means that each time the filter is changed between the options: _All_, _Unassigned_ and _Assigned_, the method is executed?
If yes, that means the every filter change makes a new request to kie server through findCaseInstance() and re-loads all the assignments all over again. Is this necessary?